numpy/ma/tests/test_core.py

```diff
@@ -1990,6 +1991,188 @@ def test_datafriendly_mul_arrays(self):
+            warnings.filterwarnings("always")
+            (x, y, xm) = self.uint8data
+            m = xm.mask
+            a = arange(10, dtype=np.uint8)
+            a[-1] = masked
+            x -= a
+            xm -= a
+            assert_equal(x, y - a)
+            assert_equal(xm, y - a)
+            assert_equal(xm.mask, mask_or(m, a.mask))
+
+            assert_equal(len(w), 0)
+
+    def test_inplace_multiplication_scalar_uint8(self):
+        # Test of inplace multiplication
+        with warnings.catch_warnings(record=True) as w:
+            warnings.filterwarnings("always")
+            (x, y, xm) = self.uint8data
+            x *= np.uint8(2)
+            assert_equal(x, y * 2)
+            xm *= np.uint8(2)
+            assert_equal(xm, y * 2)
+
+            assert_equal(len(w), 0)
+
+    def test_inplace_multiplication_array_uint8(self):
+        # Test of inplace multiplication
+        with warnings.catch_warnings(record=True) as w:
+            warnings.filterwarnings("always")
+            (x, y, xm) = self.uint8data
+            m = xm.mask
+            a = arange(10, dtype=np.uint8)
+            a[-1] = masked
+            x *= a
+            xm *= a
+            assert_equal(x, y * a)
+            assert_equal(xm, y * a)
+            assert_equal(xm.mask, mask_or(m, a.mask))
+
+            assert_equal(len(w), 0)
+
+    def test_inplace_floor_division_scalar_uint8(self):
+        # Test of inplace division
+        with warnings.catch_warnings(record=True) as w:
+            warnings.filterwarnings("always")
+            (x, y, xm) = self.uint8data
+            x = arange(10, dtype=np.uint8) * np.uint8(2)
+            xm = arange(10, dtype=np.uint8) * np.uint8(2)
+            xm[2] = masked
+            x //= np.uint8(2)
+            xm //= np.uint8(2)
+            assert_equal(x, y)
+            assert_equal(xm, y)
+
+            assert_equal(len(w), 0)
+
+    def test_inplace_floor_division_array_uint8(self):
+        # Test of inplace division
+        with warnings.catch_warnings(record=True) as w:
+            warnings.filterwarnings("always")
+            (x, y, xm) = self.uint8data
+            m = xm.mask
+            a = arange(10, dtype=np.uint8)
+            a[-1] = masked
+            x //= a
+            xm //= a
+            assert_equal(x, y // a)
+            assert_equal(xm, y // a)
+            assert_equal(xm.mask, mask_or(mask_or(m, a.mask), (a == 0)))
+
+            assert_equal(len(w), 0)
+
+    def test_inplace_division_scalar_uint8(self):
+        # Test of inplace division
+        with warnings.catch_warnings(record=True) as w:
+            warnings.filterwarnings("always")
+            (x, y, xm) = self.uint8data
+            x = arange(10, dtype=np.uint8) * np.uint8(2)
+            xm = arange(10, dtype=np.uint8) * np.uint8(2)
+            xm[2] = masked
+
+            # May get a DeprecationWarning or a TypeError.
+            # Will handle the same way.
+            try:
```

Comment:
I suppose if I stuck with this code. I could see how we could improve it. In other words, check that when both pieces of data are some type of float that we get no errors. However, I have worked on implementing a more general type check as we previously discussed and so have effectively done this.